common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1136,6 +1136,56 @@ public UTF8String replace(UTF8String search, UTF8String replace) {
     return buf.build();
   }
 
+  public UTF8String replace(UTF8String search, UTF8String replace, int collationId) {
```

Comment:
I think we now have a lot of new methods 
consider adding comments such as: https://github.com/apache/spark/pull/45791/files#r1549415554